src/components/data/datagrid/datagrid.tsx

```diff
@@ -110,6 +110,12 @@ export type DataGridProps = {
   /** References to the selected items in `objectList`, setting this preselects the items. */
   selected?: AttributeData[];
 
+  /** Can be used to specify how to compare the selected items and the items in the data grid */
+  customComparisonFunction?: (
```

Comment:
I like this approach (as I was struggling with the reference checks earlier).

There is a name/interface for a function that compares objects for ordering [`compareFn`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#comparefn). At fist I wanted to suggestion altering the name to something like `equalityChecker` however I think we could implement `compareFn` here and in the future utilize it for sorting as well, we would have to add a custom return value type for indicating not equal though (possibly false).

Motivation: this creates a base for implementing easier custom (client side) sort behavior and improves recognizablity.

Reply:
Well, on second thought. This might be odd. Lets stick with the current implementation but rename it to something more clear: suggestion: equalityChecker (brought to you by ChatGPT).